root/usr/lib64/ruby/vendor_ruby/discovery.rb

```diff
@@ -166,10 +166,10 @@ def upload(uri = discover_server, type = proxy_type, custom_facts = {})
     return
   end
   if uri.host.nil? or uri.port.nil?
-    log_err "Foreman URI host or port was not specified, cannot continue"
+    log_err "Server or proxy URI host or port was not specified, cannot continue"
```

Comment:
This and above are log output, users unlikely are facing this often, therefore I don't think it matters much. Those strings only appear in system log and only on errors.